### ISSUE TITLE
Games editing main menu styling

### ISSUE THREAD

**Issue by Extex101** [minetest/minetest#11044]
Much has been suggested for giving games access to editing the main menu, but my alternative is keeping the menu exactly how minetest ships it, but allowing them to edit the styling via something like a formspec_prepend but for the menu

**Comment by AFCMS** [comment]
Not sure its good. Main menu will become not graphically unified with this feature

**Comment by mark-wiemer** [comment]
Bumping this as it's still top of the search results, anyone know of updates here?

**Comment by appgurueu** [comment]
Part of the problem with this suggestion is that currently formspecs don't separate styles and elements clearly. Which means that if you can style the main menu, you can also modify it.

Hacks could be written e.g. to only allow style elements but then there's also the issue that this is really a crutch in the longer term - both formspecs and the current mainmenu are to be obsoleted (and really eventually we do want something like a restricted main menu API we can expose to games).

In terms of progress, it's probably worth mentioning that sfan has put some basic security for the current mainmenu env in place, which will make implementing more restrictive main menu envs easier in the future.

**Comment by mark-wiemer** [comment]
Makes sense that the Luanti launcher needs to be extremely careful allowing modification to its main menu. I think I'll ignore main menu styles for now, and if I need them, I'll ship a separate launcher :)
